awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx

```diff
@@ -513,6 +631,109 @@ function JobTemplateForm({
                 </FormCheckboxLayout>
               </FormGroup>
             </FormFullWidthLayout>
+            {enableWebhooks && (
+              <FormColumnLayout>
+                <FormGroup
+                  name="webhook_service"
+                  fieldId="webhook_service"
+                  helperTextInvalid={webhookServiceMeta.error}
+                  label={i18n._(t`Webhook Service`)}
+                >
+                  <FieldTooltip
+                    content={i18n._(t`Select a webhook service.`)}
+                  />
+                  <AnsibleSelect
+                    {...webhookServiceField}
+                    id="webhook_service"
+                    data={webhookServiceOptions}
+                    onChange={(event, val) => {
+                      webhookServiceHelpers.setValue(val);
+                      webhookUrlHelpers.setValue(
+                        `${urlOrigin}/api/v2/job_templates/${template.id}/${val}/`
+                      );
+                      if (
+                        val === webhookServiceMeta.initialValue ||
+                        val === ''
+                      ) {
+                        webhookKeyHelpers.setValue(webhookKeyMeta.initialValue);
+                        webhookCredentialHelpers.setValue(
+                          webhookCredentialMeta.initialValue
+                        );
+                      } else {
+                        webhookKeyHelpers.setValue(
+                          i18n
+                            ._(t`a new webhook key will be generated on save.`)
+                            .toUpperCase()
+                        );
+                        webhookCredentialHelpers.setValue(null);
+                      }
+                    }}
+                  />
+                </FormGroup>
+                {!jtAddMatch && (
```

Comment:
so these fields are only avaialble on edit?

Reply:
yes because in add, they have no value and are just blank fields as their are generated from the api.  It works that way in workflow job template form also.